tests/conftest.py

```diff
@@ -22,27 +22,78 @@ def requires_module(function, name, call=None):
 requires_seaborn = partial(requires_module, name="seaborn")
 
 
-def generate_cov(n_trials, n_channels):
-    """Generate a set of cavariances matrices for test purpose"""
-    rs = np.random.RandomState(1234)
+def generate_cov(n_trials, n_channels, rs, return_params=False):
+    """Generate a set of covariances matrices for test purpose"""
     diags = 2.0 + 0.1 * rs.randn(n_trials, n_channels)
     A = 2 * rs.rand(n_channels, n_channels) - 1
     A /= np.linalg.norm(A, axis=1)[:, np.newaxis]
     covmats = np.empty((n_trials, n_channels, n_channels))
     for i in range(n_trials):
         covmats[i] = A @ np.diag(diags[i]) @ A.T
-    return covmats, diags, A
+    if return_params:
+        return covmats, diags, A
+    else:
+        return covmats
+
+
+@pytest.fixture
+def rndstate():
+    return np.random.RandomState(1234)
 
 
 @pytest.fixture
-def covmats():
-    """Generate covariance matrices for test"""
-    covmats, _, _ = generate_cov(6, 3)
-    return covmats
+def get_covmats(rndstate):
+    def _gen_cov(n_trials, n_chan):
+        return generate_cov(n_trials, n_chan, rndstate, return_params=False)
+
+    return _gen_cov
 
 
 @pytest.fixture
-def many_covmats():
-    """Generate covariance matrices for test"""
-    covmats, _, _ = generate_cov(100, 3)
-    return covmats
+def get_covmats_params(rndstate):
+    def _gen_cov_params(n_trials, n_chan):
+        return generate_cov(n_trials, n_chan, rndstate, return_params=True)
+
+    return _gen_cov_params
+
+
+def get_distances():
+    distances = [
+        "riemann",
+        "logeuclid",
+        "euclid",
+        "logdet",
+        "kullback",
+        "kullback_right",
+        "kullback_sym",
+    ]
+    for dist in distances:
+        yield dist
+
+
+def get_means():
+    means = [
+        "riemann",
+        "logeuclid",
+        "euclid",
+        "logdet",
+        "identity",
+        "wasserstein",
+        "ale",
```

Comment:
What about `alm` mean?

Reply:
too slow for systematic testing. It has its own tests.